PCbuild/python.props

```diff
@@ -61,17 +61,38 @@
   <Import Project="$(ExternalProps)" Condition="$(ExternalProps) != '' and Exists('$(ExternalProps)')" />
 
   <PropertyGroup>
-    <sqlite3Dir Condition="$(sqlite3Dir) == ''">$(ExternalsDir)sqlite-3.35.5.0\</sqlite3Dir>
+    <sqlite3Dir>$(SQLITE3_DIR)</sqlite3Dir>
+    <sqlite3Dir Condition="$(sqlite3Dir) == ''">$(ExternalsDir)sqlite-3.45.1.0\</sqlite3Dir>
+    <sqlite3Dir Condition="!HasTrailingSlash($(sqlite3Dir))">$(sqlite3Dir)\</sqlite3Dir>
+    <bz2Dir>$(BZIP2_DIR)</bz2Dir>
     <bz2Dir Condition="$(bz2Dir) == ''">$(ExternalsDir)bzip2-1.0.8\</bz2Dir>
-    <lzmaDir Condition="$(lzmaDir) == ''">$(ExternalsDir)xz-5.2.2\</lzmaDir>
-    <libffiDir Condition="$(libffiDir) == ''">$(ExternalsDir)libffi-3.3.0\</libffiDir>
-    <libffiOutDir Condition="$(libffiOutDir) == ''">$(libffiDir)$(ArchName)\</libffiOutDir>
-    <libffiIncludeDir Condition="$(libffiIncludeDir) == ''">$(libffiOutDir)include</libffiIncludeDir>
+    <bz2Dir Condition="!HasTrailingSlash($(bz2Dir))">$(bz2Dir)\</bz2Dir>
+    <lzmaDir>$(LZMA_DIR)</lzmaDir>
+    <lzmaDir Condition="$(lzmaDir) == ''">$(ExternalsDir)xz-5.2.5\</lzmaDir>
+    <lzmaDir Condition="!HasTrailingSlash($(lzmaDir))">$(lzmaDir)\</lzmaDir>
+    <libffiDir>$(FFI_DIR)</libffiDir>
+    <libffiDir Condition="$(libffiDir) == ''">$(ExternalsDir)libffi\$(ArchName)\</libffiDir>
+    <libffiDir Condition="!HasTrailingSlash($(libffiDir))">$(libffiDir)\</libffiDir>
+    <libffiIncludeDir>$(libffiDir)include</libffiIncludeDir>
+    <libffiOutDir>$(libffiDir)</libffiOutDir>
+    <opensslDir>$(OPENSSL_DIR)</opensslDir>
+    <opensslIncludeDir/>
+    <opensslLibdir/>
+    <opensslOutDir Condition="$(opensslDir) == ''">$(ExternalsDir)openssl-bin-1.1.1w\$(ArchName)\</opensslOutDir>
+    <opensslLibDir Condition="$(opensslDir) == ''">$(opensslOutDir)</opensslLibDir>
+    <opensslIncludeDir Condition="$(opensslDir) == ''">$(opensslOutDir)include</opensslIncludeDir>
+
     <opensslDir Condition="$(opensslDir) == ''">$(ExternalsDir)openssl-1.1.1w\</opensslDir>
```

Comment:
Is this right?

Reply:
This only matters if it can't find the *stuff* in question, then it defaults to something it knows